### PR TITLE
Add annotations param to TfJob

### PR DESCRIPTION
**Pull request by dalfos**
**What this PR does / why we need it**:
Add annotations param to TfJob for ability to pass it to Job.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:
```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/406)

### CODE REVIEW

**Comment by jinchihe** [comment]
Hey @dalfos 
Could you please fix the pylint error in `kubeflow/fairing/deployers/tfjob/tfjob.py`? 
```
INFO|2019-10-25T23:48:14|util.py:71| ************* Module kubeflow.fairing.deployers.tfjob.tfjob
INFO|2019-10-25T23:48:14|util.py:71| C: 32, 0: Line too long (114/100) (line-too-long)
```

**Comment by dalfos** [comment]
Hi @jinchihe where can I see error in test process?

**Comment by jinchihe** [comment]
Error below.  Should be caused by test env.
```
Step #2: Fetching cluster endpoint and auth data.
Step #2: ERROR: (gcloud.container.clusters.get-credentials) ResponseError: code=404, message=Not found: projects/kubeflow-ci/zones/us-cen
tral1-a/clusters/kubeflow-ci-fairing.
Step #2: No cluster named 'kubeflow-ci-fairing' in kubeflow-ci.
Finished Step #2
ERROR
ERROR: build step 2 "gcr.io/kubeflow-ci/fairing-test:latest" failed: exit status 1
```

/retest

**Comment by jinchihe** [comment]
@dalfos The problem may can be work around by rebasing. Could you please rebase and trigger new CI testing? Thanks.

**Comment by jinchihe** [comment]
@dalfos Seems the errors are caused by testing env problem, may be resolved later. thanks.

File: kubeflow/fairing/deployers/tfjob/tfjob.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class TfJob(Job):
         training job using Kubeflow TFOperator"""
     def __init__(self, namespace=None, worker_count=1, ps_count=0,
                  chief_count=1, runs=1, job_name=constants.TF_JOB_DEFAULT_NAME, stream_log=True,
-                 labels=None, pod_spec_mutators=None, cleanup=False):
+                 labels=None, pod_spec_mutators=None, cleanup=False, annotations=None):
```
Comment (abhi-g) [inline kubeflow/fairing/deployers/tfjob/tfjob.py]: please add corresponding docstring param below line 28

Comment (dalfos) [reply kubeflow/fairing/deployers/tfjob/tfjob.py]: done

**Comment by abhi-g** [comment]
@dalfos Can you please merge upstream/master to pull in test fixes

**Comment by dalfos** [comment]
> @dalfos Can you please merge upstream/master to pull in test fixes

my fork's master already in sync with upstream

**Comment by abhi-g** [comment]
Hmm. I pushed a test fix yesterday. If you can rebase your PR on the new
upstream/master it will run CI with the updated test.

**Pull request merged**